service/history/replication/dlq_handler_test.go

```diff
@@ -110,11 +113,176 @@ func (s *dlqHandlerSuite) TearDownTest() {
 	s.mockShard.Finish(s.T())
 }
 
+func (s *dlqHandlerSuite) TestNewDLQHandler_panic() {
+	s.Panics(func() { NewDLQHandler(s.mockShard, nil) }, "Failed to initialize replication DLQ handler due to nil task executors")
+}
+
+func (s *dlqHandlerSuite) TestStart() {
+	tests := []struct {
+		name   string
+		status int32
+	}{
+		{
+			name:   "started",
+			status: common.DaemonStatusInitialized,
+		},
+		{
+			name:   "not started",
+			status: common.DaemonStatusStopped,
+		},
+	}
+
+	for _, tc := range tests {
+		s.T().Run(tc.name, func(t *testing.T) {
+			s.messageHandler.status = tc.status
+
+			s.messageHandler.Start()
+
+			s.messageHandler.Stop()
+		})
+	}
+}
+
+func (s *dlqHandlerSuite) TestStop() {
```

Comment:
what is the point of this test given that above already covers start/stop flow?

Reply:
Removed